rclpy/src/rclpy/_rclpy_logging.cpp

```diff
@@ -206,14 +206,18 @@ rclpy_logging_get_logging_directory()
 }
 
 /// Add a subordinate logger based on a logger
-void
+bool
 rclpy_logging_rosout_add_sublogger(const char * logger_name, const char * sublogger_name)
 {
   rclpy::LoggingGuard scoped_logging_guard;
   rcl_ret_t rcl_ret = rcl_logging_rosout_add_sublogger(logger_name, sublogger_name);
-  if (RCL_RET_OK != rcl_ret) {
-    rcutils_reset_error();
-    throw std::runtime_error("Failed to call rcl_logging_rosout_add_sublogger");
+  if (RCL_RET_OK == rcl_ret) {
+    return true;
+  } else if (RCL_RET_NOT_FOUND == rcl_ret) {
+    rcl_reset_error();
+    return false;
+  } else {
+    throw rclpy::append_rcl_error("Failed to call rcl_logging_rosout_add_sublogger");
```

Comment:
having exception and false looks redundant? i think for user perspective, it is not really clear the difference between having exception and false via this call.
how about keeping this `void` and if `RCL_RET_NOT_FOUND` print warning (there will no rosout) w/o exception, and everything else goes to exception?

Reply:
Using a boolean is to make sure https://github.com/ros2/rclpy/blob/059bac65707c0e605a08bcf9451738f891ed0d26/rclpy/rclpy/impl/rcutils_logger.py#L238-L239 not to update `logger.logger_sublogger_namepair` if it's false(there is no parent logger existent),
and then it's unnecessary to call `_rclpy.rclpy_logging_rosout_remove_sublogger` https://github.com/ros2/rclpy/blob/059bac65707c0e605a08bcf9451738f891ed0d26/rclpy/rclpy/impl/rcutils_logger.py#L222-L223

If there exist other errors (e.g. `RCUTILS_RET_INVALID_ARGUMENT`, etc) when calling rcutils_hash_map_get, `rclpy_logging_rosout_add_sublogger` just throw the exceptions similar to other APIs in current file.

If only printing warning for `RCL_RET_NOT_FOUND`, there exists some unnecessary actions(`add logger.logger_sublogger_namepair` and `_rclpy.rclpy_logging_rosout_remove_sublogger`).